Languages/ja/03_Function_ja/readme.md

```diff
@@ -0,0 +1,171 @@
+# WTF Solidity 超シンプル入門: 3. Function（関数）
+
+最近、Solidity の学習を再開し、詳細を確認しながら「Solidity 超シンプル入門」を作っています。これは初心者向けのガイドで、プログラミングの達人向けの教材ではありません。毎週 1〜3 レッスンのペースで更新していきます。
+
+僕のツイッター：[@0xAA_Science](https://twitter.com/0xAA_Science)｜[@WTFAcademy\_](https://twitter.com/WTFAcademy_)
+
+コミュニティ：[Discord](https://discord.gg/5akcruXrsk)｜[Wechat](https://docs.google.com/forms/d/e/1FAIpQLSe4KGT8Sh6sJ7hedQRuIYirOoZK_85miz3dw7vA1-YjodgJ-A/viewform?usp=sf_link)｜[公式サイト wtf.academy](https://wtf.academy)
+
+すべてのソースコードやレッスンは github にて公開: [github.com/AmazingAng/WTFSolidity](https://github.com/AmazingAng/WTFSolidity)
+
+-----
+
+## 関数型
+
+Solidityにおける関数の書式を示します:
+
+```solidity
+    function <function name>(<parameter types>) [internal|external] [pure|view|payable] [returns (<return types>)]
+```
+
+複雑に見えるかもしれませんが、１つ１つ分解して見てみましょう（角括弧はオプションのキーワードを示します）:
+
+
+1. `function`: 関数を記述する為に、`function`というキーワードで始める必要があります。
+
+2. `<function name>`: 関数名です。
+
+3. `(<parameter types>)`: 引数の型と名前です。
+
+4. `[internal|external|public|private]`: 関数の可視性（ビジビリティ）の修飾子です。デフォルトのビジビリティはありませんので、それぞれの関数に対してビジビリティを指定する必要があります。そしてビジビリティは4種類あるのです:
+
+   - `public`: 全てにおいて可視的です。
+
+   - `private`: このコントラクト内でのみアクセス出来ますが、派生したコントラクトでは使用出来ません。
+
+   - `external`: 別の他のコントラクトからのみ呼び出せます。しかし、コントラクト内で`f`は関数名だとすると、コントラクト内で`this.f()`で呼び出せます。 
+
+   - `internal`: コントラクト内部およびそのコントラクトから派生するコントラクトからのみアクセス出来る。
+
+    **Note 1**: `public`はデフォルトで関数のビジビリティとなっています。
+    
+    **Note 2**: `public|private|internal`は状態変数でも使用されます。パブリックな変数は値を照会する為の`getter`関数を自動的に生成します。
+    
+    **Note 3**: デフォルトでは状態変数のビジビリティは`internal`となります。
+
+5. `[pure|view|payable]`: Solidityの関数のふるまいを示すキーワードです。`payable`は理解しやすいです。`payable`関数を通して`ETH`をコントラクトに送ることが出来ます。`pure`と`view`は次の節で紹介します。
+
+6. `[returns (<return types>)]`: 変数の型と名前を返します。※返り値の名前は省略可能です。
+
+## `Pure` と `View`って一体何なんだ?
+
+私が`solidity`を学び始めた時、まったく`pure`と`view`について理解出来ませんでした。というのも、他のプログラミング言語では馴染みがないからです。`solidity`は`gas fee`があるが故に、これら2つのキーワードを追加しました。コントラクトの状態変数はブロックチェーンに保管されていて、`gas fee`はとても値段が高いです。もしこれらの状態変数を書き換えない場合、`gas`を支払う必要はありません。すなわち、`pure`と`view`の関数を呼び出す際には`gas`を支払う必要はありません。
```

Comment:
「私が」というより、「みなさんが」だったようなニュアンスな気もしますね

Reply:
「刚开始学习 solidity 时，pure 和 view 关键字可能令人费解」
→「みなさんがsolidityを学び始めた時に、pureやviewについて戸惑うと思います」
原文のままだとこれですね。どうでしょうか。

Reply:
@thurendous  CC: @AmazingAng 
**中文**
>刚开始学习 solidity 时，pure 和 view 关键字可能令人费解，因为其他编程语言中没有类似的关键字。

**英文**
>When I started learning solidity, I didn't understand pure and view at all, since they are not common in other languages. 

この差分についてどう思いますか？どちらに揃えた方が意訳になりますか？

Reply:
うん、確かに英語だとそうなるのか。どちらでもいいですわ。

Reply:
@thurendous 
OK,そしたら現行のまま保存します
ご理解よろしく！

Reply:
完了✅